code/data_structures/src/stack/balanced_expression/balanced_expression.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <stack>
+
+using namespace std;
+
+bool isOpen(char c)
+{
+    if(c == '(' || c == '{' || c == '[')
```

Comment:
How about: `return c == '(' || c == '{' || c == '[';`? You could also consider marking this function `inline`.

Reply:
As the return type is `bool` so we need to return `true` or `false` and not `1` or `0`. I have checked that the code fails in that case.